lib/widgets/new_dm_sheet.dart

```diff
@@ -0,0 +1,401 @@
+  const _SelectedUserChip({
+    required this.userId,
+    required this.unselectUser,
+  });
+
+  final int userId;
+  final void Function(int) unselectUser;
+
+  @override
+  Widget build(BuildContext context) {
+    final designVariables = DesignVariables.of(context);
+    final store = PerAccountStoreWidget.of(context);
+    final clampedTextScaler = MediaQuery.textScalerOf(context)
+      .clamp(maxScaleFactor: 1.5);
+
+    return GestureDetector(
+      onTap: () => unselectUser(userId),
+      child: DecoratedBox(
+        decoration: BoxDecoration(
+          color: designVariables.bgMenuButtonSelected,
+          borderRadius: BorderRadius.circular(3)),
+        child: Row(mainAxisSize: MainAxisSize.min, children: [
+          Avatar(userId: userId, size: clampedTextScaler.scale(22), borderRadius: 3),
+          Flexible(
+            child: Padding(
+              padding: const EdgeInsetsDirectional.fromSTEB(5, 3, 4, 3),
+              child: Text(store.userDisplayName(userId),
+                textScaler: clampedTextScaler,
+                maxLines: 1,
+                overflow: TextOverflow.ellipsis,
+                style: TextStyle(
+                  fontSize: 16,
+                  height: 16 / 16,
+                  color: designVariables.labelMenuButton)))),
+        ])));
+  }
+}
+
+class _NewDmUserList extends StatelessWidget {
+  const _NewDmUserList({
+    required this.filteredUsers,
+    required this.selectedUserIds,
+    required this.scrollController,
+    required this.onUserTapped,
+  });
+
+  final List<User> filteredUsers;
+  final Set<int> selectedUserIds;
+  final ScrollController scrollController;
+  final void Function(int userId) onUserTapped;
+
+  @override
+  Widget build(BuildContext context) {
+    final designVariables = DesignVariables.of(context);
+    final store = PerAccountStoreWidget.of(context);
+    final zulipLocalizations = ZulipLocalizations.of(context);
+
+    if (filteredUsers.isEmpty) {
+      // TODO(design): Missing in Figma.
+      return Center(
+        child: Padding(
+          padding: const EdgeInsets.all(24),
+          child: Text(
+            textAlign: TextAlign.center,
+            zulipLocalizations.newDmSheetNoUsersFound,
+            style: TextStyle(
+              color: designVariables.labelMenuButton,
+              fontSize: 16))));
+    }
+
+    return Padding(
+      padding: const EdgeInsets.symmetric(horizontal: 8),
+      child: CustomScrollView(controller: scrollController, slivers: [
+        SliverPadding(
+          padding: EdgeInsets.only(top: 8),
+          sliver: SliverSafeArea(
+            minimum: EdgeInsets.only(bottom: 8),
+            sliver: SliverList.builder(
+              itemCount: filteredUsers.length,
+              itemBuilder: (context, index) {
+                final user = filteredUsers[index];
+                final isSelected = selectedUserIds.contains(user.userId);
+
+                return Material(
```

Comment:
nit: let's pull this expression out as its own widget, say `_NewDmUserListItem`; it'll cut a few levels of indentation, and I think it's a natural abstraction boundary